src/device/device-management.ts

```diff
@@ -70,28 +103,67 @@ export async function createCameraTrack(
  * @returns A LocalTrack object or an error.
  */
 export async function createMicrophoneTrack(
-  constraints?: AudioDeviceConstraints
+  constraints?: MicrophoneConstraints
 ): Promise<LocalMicrophoneTrack> {
   let stream: MediaStream;
   try {
-    stream = await media.getUserMedia({ audio: { ...constraints } });
+    stream = await media.getUserMedia({ audio: generateAudioConstraints(constraints) });
   } catch (error) {
-    throw new WcmeError(
+    throw new WebrtcError(
       ErrorTypes.CREATE_MICROPHONE_TRACK_FAILED,
       `Failed to create microphone track ${error}`
     );
   }
-  return new LocalMicrophoneTrack(stream);
+  // See if we can just pass the track and not streams
+  return new LocalMicrophoneTrack(stream.getAudioTracks()[0]);
 }
 
 /**
  * Creates a display video track.
  *
+ * @param constraints - Display constraints for screen sharing.
  * @returns A Promise that resolves to a LocalDisplayTrack.
  */
-export async function createDisplayTrack(): Promise<LocalDisplayTrack> {
-  const stream = await media.getDisplayMedia({ video: true });
-  return new LocalDisplayTrack(stream);
+export async function createDisplayTrack(
+  constraints?: VideoConstraints
+): Promise<LocalDisplayTrack> {
+  const stream = await media.getDisplayMedia({ video: generateVideoConstraints(constraints) });
+  return new LocalDisplayTrack(stream.getVideoTracks()[0]);
+}
+
+/**
+ * Creates custom video track .
+ *
+ * @param config - Custom video track creation config.
+ * @returns A Promise that resolves to a LocalVideoTrack.
+ */
+export async function createCustomVideoTrack(
```

Comment:
not sure if we actually need these "createCustom*Track" functions... why not just add encoder config as a parameter to the LocalVideoTrack/LocalAudioTrack constructor?

Reply:
This will be more of a developer facing function , good to be more specific, If it was a internal class would have gone with the constructor way

Reply:
I agree with @marcin-bazyl , I don't actually think this is friendlier: it ends up being more confusing in my opinion because they aren't really that different, and the alternate API makes it more confusing as to which should be used.

Reply:
Was looking into this @marcin-bazyl  , When someone passes the track we can ask them to do 

let customTrack = new LocalVideoTrack(track)
customTrack.setEncoderConfig(config)

or 

let customTrack = new LocalVideoTrack(track, encoderConfig)

which will indirectly save and apply the encoderConfig

Reply:
I removed those function now @bbaldino  @marcin-bazyl

Reply:
done